dataflowengineoss/src/main/scala/io/shiftleft/dataflowengineoss/language/nodemethods/ExpressionMethods.scala

```diff
@@ -0,0 +1,32 @@
+package io.shiftleft.dataflowengineoss.language.nodemethods
+
+import io.shiftleft.codepropertygraph.generated.nodes
+import io.shiftleft.dataflowengineoss.semanticsloader.{FlowSemantic, Semantics}
+import io.shiftleft.semanticcpg.language.{NoResolve, toExpression}
+import overflowdb.traversal.NodeOps
+
+class ExpressionMethods[NodeType <: nodes.Expression](val node: NodeType) extends AnyVal {
+
+  def isUsed(implicit semantics: Semantics): Boolean = {
```

Comment:
Allow expressions to be inspected to see if they are used/defined according to the semantic.